hummingbot/connector/exchange/bloxroute_openbook/bloxroute_openbook_exchange.py

```diff
@@ -0,0 +1,485 @@
+    BloxrouteOpenbookProvider,
+)
+from hummingbot.connector.exchange.bloxroute_openbook.bloxroute_openbook_utils import (
+    order_type_to_blxr_order_type,
+    trade_type_to_side,
+)
+from hummingbot.connector.exchange_py_base import ExchangePyBase
+from hummingbot.connector.trading_rule import TradingRule
+from hummingbot.core.api_throttler.async_throttler import AsyncThrottler
+from hummingbot.core.data_type.common import OrderType, TradeType
+from hummingbot.core.data_type.in_flight_order import InFlightOrder, OrderState, OrderUpdate, TradeUpdate
+from hummingbot.core.data_type.order_book import OrderBook
+from hummingbot.core.data_type.order_book_tracker_data_source import OrderBookTrackerDataSource
+from hummingbot.core.data_type.trade_fee import DeductedFromReturnsTradeFee, TradeFeeBase
+from hummingbot.core.data_type.user_stream_tracker_data_source import UserStreamTrackerDataSource
+from hummingbot.core.network_iterator import NetworkStatus
+from hummingbot.core.web_assistant.auth import AuthBase
+from hummingbot.core.web_assistant.connections.data_types import RESTMethod
+from hummingbot.core.web_assistant.web_assistants_factory import WebAssistantsFactory
+
+if TYPE_CHECKING:
+    from hummingbot.client.config.config_helpers import ClientConfigAdapter
+
+
+class BloxrouteOpenbookExchange(ExchangePyBase):
+    def __init__(
+        self,
+        client_config_map: "ClientConfigAdapter",
+        bloxroute_auth_header: str,
+        solana_wallet_private_key: str,
+        trading_pairs: Optional[List[str]] = None,
+        trading_required: bool = True,
+    ):
+        """
+        :param bloxroute_auth_header: The bloXroute Labs authorization header to connect with the Solana Trader API
+        :param solana_wallet_private_key: The secret key for your Solana wallet
+        :param trading_pairs: The market trading pairs used for a strategy
+        :param trading_required: Whether actual trading is needed
+        """
+
+        self.logger().info("Creating bloXroute exchange")
+
+        self._trading_required = trading_required
+        self._auth_header = bloxroute_auth_header
+        self._sol_wallet_private_key = solana_wallet_private_key
+        kp = load_private_key(self._sol_wallet_private_key)
+        self._sol_wallet_public_key = str(kp.public_key)
+
+        self._order_id_mapper: Dict[str, int] = {}  # maps Hummingbot to bloXroute order id
+        self._open_orders_address_mapper: Dict[str, str] = {}  # maps trading pair to open orders address
+
+        self._token_accounts: Dict[str, str] = {}
+        self._trading_pairs = trading_pairs
+        asyncio.create_task(self._initialize_token_accounts())
+
+        self._order_manager: BloxrouteOpenbookOrderDataManager = BloxrouteOpenbookOrderDataManager(
+            self._provider, self._trading_pairs, self._sol_wallet_public_key
+        )
+        asyncio.create_task(self._initialize_order_manager())
```

Comment:
I don't think this is actually blocking, but that is the intended use?

Reply:
I don't think it is possible to block in the init function - but we have an `asyncio.Event` to wait for the order manager and I added something to the `status_dict` for the `token_accounts`

Reply:
We actually await the task now in the BloxrouteOpenbookProvider